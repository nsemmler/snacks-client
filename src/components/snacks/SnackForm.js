import React from 'react';
import {
  CustomInput,
  Form,
  FormGroup,
  Label,
  Input,
  Col,
  Button,
  InputGroup,
  InputGroupAddon
} from 'reactstrap';

const SnackForm = ({handleEditSnack, handleAddSnack, singleSnack}) => {

  const snack = {
    id: singleSnack
      ? singleSnack.id
      : "",
    name: singleSnack
      ? singleSnack.name
      : "",
    description: singleSnack
      ? singleSnack.description
      : "",
    price: singleSnack
      ? Number(singleSnack.price)
      : "",
    img: singleSnack
      ? singleSnack.img
      : "",
    is_perishable: singleSnack
      ? singleSnack.is_perishable
      : ""
  }

  return (<Form className="form-horizontal well" onSubmit={event => {
      event.preventDefault();
      snack.id
        ? handleEditSnack(snack.id, event.target.title.value, event.target.description.value, event.target.price.value, event.target.img.value, (event.target.is_perishable.value || false))
        : handleAddSnack(event.target.title.value, event.target.description.value, event.target.price.value, event.target.img.value, (event.target.is_perishable.value || false));
    }}>

    <FormGroup>
      <Label for="title">Snack Name</Label>
      <Input type="textarea" defaultValue={snack.name} name="title" id="title" required="required"/>
    </FormGroup>

    <FormGroup>
      <Label for="description">Snack Description</Label>
      <Input type="textarea" defaultValue={snack.description} name="description" id="description" required="required"/>
    </FormGroup>
    <FormGroup>
      <Label for="url">Url</Label>
      <Input type="url" name="img" defaultValue={snack.img} id="img" placeholder="url" required="required"/>
    </FormGroup>

    <FormGroup>
      <Label for="price">Snack Price?</Label>
      <InputGroup>
        <InputGroupAddon addonType="prepend">$</InputGroupAddon>
        <Input name="price" id="name" defaultValue={snack.price} placeholder="Amount" type="number" required="required"/>
        <InputGroupAddon addonType="append">.00</InputGroupAddon>
      </InputGroup>
    </FormGroup>
    <FormGroup>
      <Label for="is_perishable">Is your snack perishable?</Label>
      <div>
        <CustomInput value="true" type="radio" id="true" name="is_perishable" label="Yes" inline="inline"/>
        <CustomInput value="false" type="radio" id="false" name="is_perishable" label="No" inline="inline"/>
      </div>
    </FormGroup>

    <FormGroup check="check">
      <Col sm={{
          size: 10,
          offset: 0
        }}>
        <Button>{
            snack.id
              ? `Edit`
              : `Add`
          }</Button>
      </Col>
    </FormGroup>
  </Form>);
}

export default SnackForm
