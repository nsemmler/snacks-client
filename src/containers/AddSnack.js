import React, {Component} from 'react'
import {connect} from 'react-redux'
import {addSnack} from '../state/actions'
import {bindActionCreators} from 'redux'
import SnackForm from '../components/snacks/SnackForm'
import Jumbo from '../components/shared/Jumbotron'

class AddSnack extends Component {
  handleAddSnack = (title, description, price, img, is_perishable) => {
    this.props.addSnack(title, description, price, img, is_perishable)
    this.props.history.push('/snacks')
  }

  render() {
    const jumboStyle = {
      height: {
        height: "30vh"
      },
      title: 'Add Snack',
      subtitle: `Can't wait to see what you come up with..`
    }

    return (<section>
      <Jumbo props={jumboStyle}/>
      <div id="form">
        <SnackForm handleAddSnack={this.handleAddSnack}/>
      </div>
    </section>)
  }
}

const mapStateToProps = ({singleSnack}) => ({singleSnack})
const mapDispatchToProps = dispatch => bindActionCreators({
  addSnack
}, dispatch)

export default connect(mapStateToProps, mapDispatchToProps)(AddSnack)
