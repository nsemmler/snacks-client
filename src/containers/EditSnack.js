import React, {Component} from 'react'
import {connect} from 'react-redux'
import {editSnack, getOneSnax} from '../state/actions'
import {bindActionCreators} from 'redux'
import SnackForm from '../components/snacks/SnackForm'
import Jumbo from '../components/shared/Jumbotron'

class EditSnack extends Component {

  componentDidMount() {
    const snackId = this.props.match.params.id
    this.props.getOneSnax(snackId)
  }

  handleEditSnack = (snackId, title, description, price, img, is_perishable) => {
    this.props.editSnack(snackId, title, description, price, img, is_perishable)
    this.props.history.push(`/snack/${snackId}`)
  }

  render() {

    const jumboStyle = {
      height: {
        height: "30vh"
      },
      title: 'Edit Snack',
      subtitle: ''
    }

    return (<section>
      <Jumbo props={jumboStyle}/>
      <div id="form">
        <SnackForm singleSnack={this.props.singleSnack} handleEditSnack={this.handleEditSnack}/>
      </div>
    </section>)
  }
}

const mapStateToProps = ({singleSnack}) => ({singleSnack})
const mapDispatchToProps = dispatch => bindActionCreators({
  editSnack,
  getOneSnax
}, dispatch)

export default connect(mapStateToProps, mapDispatchToProps)(EditSnack)
