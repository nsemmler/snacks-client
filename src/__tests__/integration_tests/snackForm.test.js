import React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'
import SnackForm from '../../components/snacks/SnackForm'
import EditSnack from '../../containers/EditSnack'
import AddSnack from '../../containers/AddSnack'

describe('Snack Form Component', () => {
  test('renders consistently', () => {
    const handleEditSnack = () => { EditSnack.handleEditSnack }
    const handleAddSnack = () => { AddSnack.handleAddSnack }
    const singleSnack = {
      id: 1,
      name: "Pork Rinds",
      description: "Mauris lacinia sapien quis libero. Nam dui. Proin leo odio, porttitor id, consequat in, consequat ut, nulla. Sed accumsan felis.",
      price: 8,
      img: "https://az808821.vo.msecnd.net/content/images/thumbs/0000398_salt-pepper-pork-rinds-2-oz_560.jpeg",
      is_perishable: true
    }

    const form = renderer.create(
      <MemoryRouter>
        <SnackForm handleEditSnack={ handleEditSnack } handleAddSnack={ handleAddSnack } singleSnack={ singleSnack } />
      </MemoryRouter>
    ).toJSON()
    expect(form).toMatchSnapshot()
  })
})
