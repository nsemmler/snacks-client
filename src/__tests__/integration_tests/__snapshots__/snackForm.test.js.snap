// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snack Form Component renders consistently 1`] = `
<form
  className="form-horizontal well"
  onSubmit={[Function]}
>
  <div
    className="position-relative form-group"
  >
    <label
      className=""
      htmlFor="title"
    >
      Snack Name
    </label>
    <textarea
      className="form-control"
      defaultValue="Pork Rinds"
      id="title"
      name="title"
      required="required"
    />
  </div>
  <div
    className="position-relative form-group"
  >
    <label
      className=""
      htmlFor="description"
    >
      Snack Description
    </label>
    <textarea
      className="form-control"
      defaultValue="Mauris lacinia sapien quis libero. Nam dui. Proin leo odio, porttitor id, consequat in, consequat ut, nulla. Sed accumsan felis."
      id="description"
      name="description"
      required="required"
    />
  </div>
  <div
    className="position-relative form-group"
  >
    <label
      className=""
      htmlFor="url"
    >
      Url
    </label>
    <input
      className="form-control"
      defaultValue="https://az808821.vo.msecnd.net/content/images/thumbs/0000398_salt-pepper-pork-rinds-2-oz_560.jpeg"
      id="img"
      name="img"
      placeholder="url"
      required="required"
      type="url"
    />
  </div>
  <div
    className="position-relative form-group"
  >
    <label
      className=""
      htmlFor="price"
    >
      Snack Price?
    </label>
    <div
      className="input-group"
    >
      <div
        className="input-group-prepend"
      >
        <span
          className="input-group-text"
        >
          $
        </span>
      </div>
      <input
        className="form-control"
        defaultValue={8}
        id="name"
        name="price"
        placeholder="Amount"
        required="required"
        type="number"
      />
      <div
        className="input-group-append"
      >
        <span
          className="input-group-text"
        >
          .00
        </span>
      </div>
    </div>
  </div>
  <div
    className="position-relative form-group"
  >
    <label
      className=""
      htmlFor="is_perishable"
    >
      Is your snack perishable?
    </label>
    <div>
      <div
        className="custom-radio custom-control custom-control-inline"
      >
        <input
          className="custom-control-input"
          id="true"
          name="is_perishable"
          type="radio"
          value="true"
        />
        <label
          className="custom-control-label"
          htmlFor="true"
        >
          Yes
        </label>
      </div>
      <div
        className="custom-radio custom-control custom-control-inline"
      >
        <input
          className="custom-control-input"
          id="false"
          name="is_perishable"
          type="radio"
          value="false"
        />
        <label
          className="custom-control-label"
          htmlFor="false"
        >
          No
        </label>
      </div>
    </div>
  </div>
  <div
    className="position-relative form-check"
  >
    <div
      className="col-sm-10 offset-sm-0"
    >
      <button
        className="btn btn-secondary"
        onClick={[Function]}
        type={undefined}
      >
        Edit
      </button>
    </div>
  </div>
</form>
`;
